src/number/n_GF.jl

```diff
@@ -384,10 +384,10 @@ system with given characteristic, degree and string. If this is not the
 desired behaviour, one can pass `false` for the optional `cached` parameter.
 """
 function FiniteField(p::Int, n::Int, S::String; cached=true)
-   n >= 16 || p >= 2^8 && throw(DomainError())
-   !Nemo.isprime(Nemo.fmpz(p)) && throw(DomainError())
-   n*log(p) >= 20*log(2) && throw(DomainError())
-   p^n >= 2^16 && throw(DomainError())
+   p >= 2^8 && throw(DomainError(p, "p must be < 256"))
+   !Nemo.isprime(Nemo.fmpz(p)) && throw(DomainError(p, "p must be prime"))
```

Comment:
off-topic (as the code was here already; so I do not expect this to be addressed here, but am interested in feedback): I haven't measured it, but I am concerned about the overhead of this check (but that may just be because I had to worry about it in GAP). But since we are only looking at values < 256 anyway, how about adding something like `const small_primes = BitSet(filter(x->Nemo.isprime(Nemo.fmpz(x)), 1:255))` to the top level, then here we can just say
```suggestion
   p in small_primes || throw(DomainError(p, "p must be prime"))
```
Benchmarking:
```
julia> @benchmark Nemo.isprime(Nemo.fmpz(17))
BenchmarkTools.Trial:
  memory estimate:  16 bytes
  allocs estimate:  1
  --------------
  minimum time:     28.111 ns (0.00% GC)
  median time:      29.848 ns (0.00% GC)
  mean time:        100.676 ns (35.81% GC)
  maximum time:     249.966 μs (53.27% GC)
  --------------
  samples:          10000
  evals/sample:     992

julia> @benchmark 17 in small_primes
BenchmarkTools.Trial:
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     2.034 ns (0.00% GC)
  median time:      2.044 ns (0.00% GC)
  mean time:        2.065 ns (0.00% GC)
  maximum time:     18.254 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1000
```